homeassistant/components/dlna_dms/dms.py

```diff
@@ -0,0 +1,715 @@
+            except UpnpError as err:
+                self._ssdp_connect_failed = True
+                LOGGER.warning(
+                    "Failed connecting to recently alive device at %s: %r",
+                    self.location,
+                    err,
+                )
+
+    async def async_config_update_listener(
+        self, hass: HomeAssistant, entry: ConfigEntry
+    ) -> None:
+        """Handle changes to the config entry associated with this entity."""
+        # Update DLNA source list when the source_id changes
+        new_source_id = entry.options[CONF_ENTITY_ID]
+        if new_source_id != self.source_id:
+            domain_data = get_domain_data(hass)
+            async with domain_data.lock:
+                if new_source_id in domain_data.sources:
+                    LOGGER.error("New source_id (%s) is already in use", new_source_id)
+                    return
+                del domain_data.sources[self.source_id]
+                self.source_id = new_source_id
+                domain_data.sources[self.source_id] = self
+
+    # Device connection/disconnection
+
+    async def device_connect(self) -> None:
+        """Connect to the device now that it's available."""
+        LOGGER.debug("Connecting to device at %s", self.location)
+
+        async with self._device_lock:
+            if self._device:
+                LOGGER.debug("Trying to connect when device already connected")
+                return
+
+            if not self.location:
+                LOGGER.debug("Not connecting because location is not known")
+                return
+
+            domain_data = get_domain_data(self.hass)
+
+            # Connect to the base UPNP device
+            upnp_device = await domain_data.upnp_factory.async_create_device(
+                self.location
+            )
+
+            # Create profile wrapper
+            self._device = DmsDevice(upnp_device, domain_data.event_handler)
+
+            # Update state variables. We don't care if they change, so this is
+            # only done once, here.
+            await self._device.async_update()
+
+    async def device_disconnect(self) -> None:
+        """Destroy connections to the device now that it's not available.
+
+        Also call when removing this entity from hass to clean up connections.
+        """
+        async with self._device_lock:
+            if not self._device:
+                LOGGER.debug("Disconnecting from device that's not connected")
+                return
+
+            LOGGER.debug("Disconnecting from %s", self._device.name)
+
+            self._device = None
+
+    # Entity properties
+
+    @property
+    def available(self) -> bool:
+        """Device is available when we have a connection to it."""
+        return self._device is not None and self._device.profile_device.available
+
+    @property
+    def unique_id(self) -> str:
+        """Get the unique ID for this entity, which is the USN."""
```

Comment:
Whyn ot call this `device_id` ?

Reply:
Dropped in favour of using the `config_entry.unique_id` in `DlnaDmsData`, which makes much more sense in context.